avr-hal-generic/src/usart_spi.rs

```diff
@@ -0,0 +1,132 @@
+//! MSPIM Implimentation
+use crate::{port::PinOps, spi};
+
+// This module just impliments a macro for SpiOps, since underlyingly, the Spi type can still be used since it just needs SpiOps
+
+/// Dummy Pin for MPSPIM
+pub struct UsartSPIDummyPin;
+
+impl PinOps for UsartSPIDummyPin {
+    type Dynamic = todo!();
+
+    fn into_dynamic(self) -> Self::Dynamic {
+        todo!()
+    }
```

Comment:
I found this in port.rs, gave it a try and at least it built. What do you think?

```suggestion
    type Dynamic = Self;

    fn into_dynamic(self) -> Self::Dynamic {
        self
    }
```

Reply:
From my side this is fine.  This pin-type is non-functional anyway - it just exists to bypass API problems.

Reply:
`Spi` only takes the CS pin to enforce a hardware restriction, it actually doesn't do anything with the CS pin related to "CS" operation at all.  This is an unfortunate side-effect of the AVR SPI peripheral design.  You can read up more on this in #27.

Actually, I had the same thought as you here:  We need the Dummy Pin to make the type-system happy, but this should be hidden from the user if at all possible.  The easiest way to hide it is by adding a new constructor, like the one you showcased here.  I think that's the best option.

Reply:
Thanks for the reference, I did not know about this behaviour 🤦‍♂️ 

My remaining issue with the separate constructor is how to enforce that the user passes actual USART peripheral? Because, as it stands, the following compiles...

```Rust
let mut spi = spi::Spi::new_from_usart(
    dp.SPI,
    pins.pb1.into_output(),
    pins.pb2.into_output(),
    pins.pb3.into_pull_up_input(),
    spi::Settings::default(),
);
```

Reply:
Hm, can we add a trait bound on `UsartOps` maybe?  If not, I'd introduce a new marker trait `UsartSpiDevice` that is implemented alongside `SpiOps` for USART peripherals.  We can then add a trait bound on this marker trait to the new constructor.

Reply:
Adding a trait bound on `UsartOps` seems to do it. I'll test on hardware and, if everything goes well, make a PR as you advised.